terra_layer/management/commands/layer_load_dump.py

```diff
@@ -44,18 +44,52 @@ def handle(self, *args, **options):
             except Field.DoesNotExist:
                 raise
 
+        parts = data["name"].split("/")
+        layer_name = parts.pop()
+
         # Try to find already existing layer
         try:
-            _, layer_name = data["name"].rsplit("/", 1)
-            group = LayerGroup.objects.get(view=data["view"], label=data["group"])
-            layer = Layer.objects.get(group=group, name=layer_name)
+            layer = Layer.objects.get(uuid=data["uuid"])
+            exists = True
         except Exception:
             layer = Layer()
+            exists = False
+
+        del data["group"]  # Remove group as we compute it later
+        data["name"] = layer_name
 
-        srlz = LayerSerializer(instance=layer, data=data)
+        srlz = LayerDetailSerializer(instance=layer, data=data)
         try:
             srlz.is_valid(raise_exception=True)
         except Exception as e:
-            raise CommandError(f"A validation error occured with data: {e}")
+            raise CommandError(f"A validation error occurred with data: {e}")
 
         srlz.save()
+
+        scene = Scene.objects.get(id=data["view"])
+        current_node = scene.tree
+
+        # Here we insert layer in tree if not previously existing
+        if not exists:
+            # Add the layer in scene tree
+            # Here we assume that missing groups are added at first position of current node
+            # We create missing group with default exclusive group configuration (should be corrected later if necessary)
+            for part in parts:
+                found = False
+                for g in current_node:
```

Comment:
Can you rename the variable `g`, please ?

Reply:
Done